crates/fetch_queue/src/fetch_task.rs

```diff
@@ -0,0 +1,52 @@
+use std::time::Duration;
+
+use kitsune2_api::fetch::{DynFetchQueue, FetchTaskT};
+
+/// Configuration for [`FetchTask`].
+pub struct FetchTaskConfig {
+    /// How long to pause after sending a fetch request, before the next attempt.
+    pub pause_between_runs: u64, // in ms
+    /// Maximum number of ops to request from a remote.
+    pub max_ops_to_request: usize,
```

Comment:
I think we need to come at this from a slightly different angle.

We don't know the size of any individual ops we will be requesting. We also don't want to send so many hashes that the hashes themselves are using a significant part of our bandwidth for the remote to only be able to respond with data to a handful of them and then we need to request them again anyways. We don't know if a particular remote will be responsive or not, and if the *are* responsive, we don't know if they might respond (no, I'm too busy right now).

I suggest, config wise, something like:

```rust
struct Config {
    /// How many parallel fetch requests for actual data we should have at once. Default: 2.
    parallel_request_count: u8,
    /// Max hash count to request from one peer at a time. Default: 16.
    max_hash_count: u8,
    /// Max byte count that we want to get in a single request. Default: 10MiB.
    /// This will be sent to the remote as part of the request. If the data of the requested ops
    /// is larger than this amount, they will send back as the ACK the count of hashes
    /// that they will be responding with that is under this byte count.
    max_byte_count: u32,
}
```

Reply:
I like this suggestion as opening up possibility of conductor-level bandwidth constraints